code/iaas/agent-server/src/main/java/io/cattle/platform/agent/server/service/impl/AgentServiceImpl.java

```diff
@@ -39,6 +39,11 @@
     public static final Set<String> GOOD_AGENT_STATES = new HashSet<String>(Arrays.asList(CommonStatesConstants.ACTIVATING, CommonStatesConstants.ACTIVE,
             AgentConstants.STATE_RECONNECTING));
 
+    public static final Set<String> REMOVED_AGENT_STATES = new HashSet<String>(Arrays.asList(
+            CommonStatesConstants.REMOVED, CommonStatesConstants.REMOVING,
+            CommonStatesConstants.DEACTIVATING, CommonStatesConstants.INACTIVE, CommonStatesConstants.PURGED,
```

Comment:
Not fixing it will result in the host being stuck in deactivating state for longer as we are dropping agent events if the agent is in this state. I can introduce a new exception.